src/ring_theory/chain_of_divisors.lean

```diff
@@ -232,11 +259,49 @@ begin
   exact ne_zero_of_dvd_ne_zero hn (subtype.prop (d ⟨c₁ 1 ^ s, _⟩))
 end
 
-variables [decidable_rel ((∣) : associates M → associates M → Prop)]
- [decidable_rel ((∣) : associates N → associates N → Prop)]
+lemma map_prime_of_monotone_equiv [decidable_eq (associates M)]
+  {m p : associates M} {n : associates N} (hn : n ≠ 0) (hp : p ∈ normalized_factors m)
+  (d : {l : associates M // l ≤ m} ≃o {l : associates N // l ≤ n}) :
```

Comment:
@YaelDillies, better or worse as
```suggestion
  (d : set.Iic m ≃o set.Iic n) :
```

Reply:
Hmm... that's a tough one. I would say `Iic` because we have some API around the equivalence between `ℕ` and `Ici (a : ℕ)`, although I cannot find it anymore now.

Reply:
I'm happy to make the change, but would it be worth waiting for this PR to be done, and then refactor the whole file, since a previous PR using `(d : {l : associates M // l ≤ m} ≃o {l : associates N // l ≤ n})` has already made its way into mathlib, just to keep this PR a bit less cluttered ?

Reply:
Never mind, the fix actually didn't need any work!